### PR TITLE
replace regular search with partial_search speeds up the cookbook

### PR DESCRIPTION
**Pull request by jvanlieshout**

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -26,12 +24,20 @@
...
 unless node['mremoteng']['shared_config_dir'].nil?
-  hosts = search(:node, 'name:*')
+  hosts = partial_search(:node, 
+                         "name:*",
+                         :keys => { 
+                           'hostname' => ['hostname'],
+                           'name' => ['name'],
+                           'os' => ['os'],
+                           'domain' => ['domain'],
+                           'env' => ['chef_environment']
```
Comment (svanharmelen) [inline recipes/default.rb]: Why would you change the `chef_environment` key to just `env` here? Is there any reason to do that? If you leave it to be just `chef_environment` I think the code is more clear and less changes are needed.

**Comment by svanharmelen** [comment]
Did you benchmark/test if the change does actually improve the speed of the search it does? Because in this case (this type of search) I doubt if you gain any performance by using a partial_search. So if the performance is not impacted, this only adds a dependency which in that case doesn't really make sense right?

**Comment by jvanlieshout** [comment]
I have tested it and it saves quite some time. I will do some tests to provide the numbers here.

**Comment by svanharmelen** [comment]
Don't bother, I believe you :smiley: So one inline comment is left....

**Comment by jvanlieshout** [comment]
I couldn't help benchmarking it just out of curiosity. The difference is even bigger then I thought :wink:.

``` ruby
require "benchmark"
 => true

time = Benchmark.measure do
hosts = search(:node, 'name:*')
end
 => 137.719000  13.031000 150.750000 (152.015698)
puts time
137.719000  13.031000 150.750000 (152.015698)
 => nil

time = Benchmark.measure do
hosts = partial_search(:node,
                          "name:*",
                          :keys => {
                          'hostname' => ['hostname'],
                          'name' => ['name'],
                          'os' => ['os'],
                          'domain' => ['domain'],
                          'env' => ['chef_environment'],
                          'ipaddress' => ['ipaddress']
                          })
end
 =>   0.016000   0.000000   0.016000 (  0.312500)
puts time
  0.016000   0.000000   0.016000 (  0.312500)
 => nil
```

By the way, what exactly do you mean with?

> So one inline comment is left....

**Comment by svanharmelen** [comment]
I also left a comment inline on one of your commits...

```
Why would you change the chef_environment key to just env here? Is there any reason to do that?
If you leave it to be just chef_environment I think the code is more clear and less changes are needed.
```

And for the benchmark... Now wait 20 minutes, and do it again but than in a different order (the partial search first ad the full search second. Does these give the same result??

**Comment by jvanlieshout** [comment]
Here you go. Absolutely agree that this is clearer.

**Comment by svanharmelen** [comment]
Sorry to be nagging, but please revert the changes to the template file (templates/default/confCons.xml.erb) made in your last commit. The rest looks great now!

**Comment by jvanlieshout** [comment]
Sorry that this has fallen behind but reverting the changes to the template will brake the cookbook... In that same commit I've renamed the variables parsed to the template so I had to change it in the template as well. Or do i miss understand your comment?

**Comment by svanharmelen** [comment]
Merged manually...

**Pull request closed without merging**